src/elements/content-preview/ContentPreview.js

```diff
@@ -362,11 +362,16 @@ class ContentPreview extends React.PureComponent<Props, State> {
      * @return {void}
      */
     componentDidUpdate(prevProps: Props, prevState: State): void {
-        const { token } = this.props;
-        const { token: prevToken } = prevProps;
+        const { token, previewExperiences } = this.props;
+        const { token: prevToken, previewExperiences: prevPreviewExperiences } = prevProps;
         const { currentFileId } = this.state;
         const hasFileIdChanged = prevState.currentFileId !== currentFileId;
         const hasTokenChanged = prevToken !== token;
+        const haveExperiencesChanged = prevPreviewExperiences !== previewExperiences;
+
+        if (haveExperiencesChanged && this.preview && this.preview.updateExperiences) {
```

Comment:
Should we do this after the other logic in this function? Otherwise it seems like we may risk operating on a stale `preview` instance.

Reply:
Yes, that makes sense 👍